### PR TITLE
EVA-1793 - Changes to support Spring Boot 2

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**
Changes to support Spring Boot 2 MongoDB object models/methods
 - ClusteringWriterTest.java
 - ClusteringVariantStepConfigurationTest.java
 - ClusteredVariantsRestControllerTest.java
 - HumanMongoConfiguration.java
 - MongoConfiguration.java
 - SubmittedVariantAccessioningRepositoryTest.java
 - SubmittedVariantAccessioningServiceTest.java

Use MongoDBBulkWriteException instead of BulkOperationException
 - ClusteringWriter.java
 - MongoBulkWriteExceptionUtils.java
 - DbsnpClusteredVariantOperationWriter.java
 - DbsnpClusteredVariantWriter.java
 - DbsnpJsonClusteredVariantsWriter.java
 - DbsnpSubmittedVariantOperationWriter.java
 - DbsnpSubmittedVariantWriter.java
 - DbsnpVariantsWriter.java
 - MergeOperationBuilder.java
 - DeprecationWriter.java

Expect DuplicateKeyException instead of BulkOperationException for accession collisions
 - DbsnpClusteredVariantWriterTest.java
 - DbsnpSubmittedVariantWriterTest.java

SubmittedVariantAccessioningRepositoryImpl.java - Use Criteria object for "and" conditions and make the findByAccessionGreaterThanEqualAndAccessionLessThanEqual method public so that it can be called from the tests

Allow bean definition overriding in properties files across all projects

Adapt to different job resume semantics in Spring Boot 2
 - DbsnpImportVariantsJobLauncherCommandLineRunner.java

Spring Boot 2 does not allow "with" method with null sort parameter on a MongoQuery object
 - MongoDbCursorItemReader.java

Adapt to Spring 2 RestTemplateAutoConfiguration behavior and package paths
 - ApplicationConfiguration.java (eva-accession-ws)
 - EvaAccessionApplication.java (eva-accession-ws)

### CODE REVIEW

File: eva-accession-core/src/test/resources/properties/ss-accession-test.properties [hunk 1]
```diff
@@ -10,12 +10,14 @@ accessioning.monotonic.test-rs.blockStartValue=3000000000
...
-spring.data.mongodb.host=|eva.mongo.host.test|
+spring.data.mongodb.host=localhost
 spring.data.mongodb.password=
 spring.data.mongodb.port=27017
+# See https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.1-Release-Notes#bean-overriding
+spring.main.allow-bean-definition-overriding=true
 
 human.mongodb.database=human-variants-test
-human.mongodb.host=|eva.mongo.host.test|
+human.mongodb.host=localhost
```
[the remark this answers is not in the document]
Comment (jmmut) [reply eva-accession-core/src/test/resources/properties/ss-accession-test.properties]: let us know how can we document this better from your point of view.

[the remark this answers is not in the document]
Comment (sundarvenkata-EBI) [reply eva-accession-core/src/test/resources/properties/ss-accession-test.properties]: I replaced it for testing but then left it as it is... Will fix this...

File: eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/batch/io/DbsnpSubmittedVariantWriterTest.java [hunk 2]
```diff
@@ -130,7 +130,7 @@ public void failsOnDifferentTaxonomies() throws Exception {
         DbsnpSubmittedVariantEntity secondVariant = new DbsnpSubmittedVariantEntity(
                 EXPECTED_ACCESSION_2, hashingFunction.apply(submittedVariant2), submittedVariant2, 1);
 
-        thrown.expect(BulkOperationException.class);
+        thrown.expect(DuplicateKeyException.class);
```
Comment (andresfsilva) [inline eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/batch/io/DbsnpSubmittedVariantWriterTest.java]: consider using `assertThrows`

**Comment by sundarvenkata-EBI** [comment]
As agreed with @jmmut  and @andresfsilva, will rebase this PR on top of command line runner PR#203 and look for any further Spring 2 specific changes to be made...

**Comment by sundarvenkata-EBI** [comment]
Thankfully, I only had to update one test property to make the clustering pipeline tests run successfully. I will handle the resumability as part of EVA-1932.

**Pull request merged**